### PR TITLE
Open options page on installation. 

### PR DESCRIPTION
**Pull request by suryatech**

### CODE REVIEW

File: src/lib/main.js [hunk 1]
```diff
@@ -59,10 +59,82 @@ tabs.on('ready', function(tab) {
...
-    if (options.loadReason === 'install') {
-        tabs.open(data.url('main/install.html'));
-    };
-};
+    // When the extension is first installed, take the user to the options
+    // page like in the chrome extensions and have him fill the settings.
+    // Check for the savedPrefs flag every single time the extensions
+    // starts up. Set the key only when the user enters a valid key for the 
+    // first time. 
+    if (options.loadReason === 'install' || options.loadReason === 'startup' || options.loadReason === 'enable') {
```
Comment (mitechie) [inline src/lib/main.js]: Can we indent a bit here and keep the line length down below 80 please?

Comment (suryatech) [inline src/lib/main.js]: Okay.

File: src/lib/preferences.js [hunk 2]
```diff
@@ -4,9 +4,14 @@ console.log('preferences.js');
...
-    prefs.on("", onPrefChange);
+    
+    // Add a listener only when the following changes
+    // Changes in cache_content key don't really matter.
+    prefs.on("api_key", onPrefChange);
+    prefs.on("api_url", onPrefChange);
+    prefs.on("api_username", onPrefChange);
     prefs.on('sync', onSyncBmarks);
-
+
```
Comment (mitechie) [inline src/lib/preferences.js:14]: looks like some extra whitespace here.

File: src/lib/preferences.js [hunk 3]
```diff
@@ -4,9 +4,14 @@ console.log('preferences.js');
...
+    // Add a listener only when the following changes
+    // Changes in cache_content key don't really matter.
+    prefs.on("api_key", onPrefChange);
+    prefs.on("api_url", onPrefChange);
+    prefs.on("api_username", onPrefChange);
     prefs.on('sync', onSyncBmarks);
-
+    
     // TODO
     // hook into /api/v1/{username}/ping to check settings
```
Comment (mitechie) [inline src/lib/preferences.js]: we can remove this comment now that the code does this correct?

**Pull request merged**